generative_data_prep/data_prep/data_prep.py

```diff
@@ -97,15 +101,33 @@ def data_prep_main(
         apply_chat_template,
     )
 
+    error_log_path = os.path.join(error_log_dir, f"{os.path.splitext(os.path.basename(input_file))[0]}.log")
+
     dump_categories = category_to_id is not None
 
     with Hdf5FileBuffer(output_file, max_seq_length, dump_categories) as hdf5_text_buffer:
         with open(input_file, "r") as reader:
             for line in reader:
-                hdf5_text_buffer.write(article_tokenizer(line))
-                if num_tokenized_articles_lock is not None and num_tokenized_articles is not None:
-                    with num_tokenized_articles_lock:
-                        num_tokenized_articles.value += 1
+                try:
+                    hdf5_text_buffer.write(article_tokenizer(line))
+                    if num_tokenized_articles_lock is not None and num_tokenized_articles is not None:
+                        with num_tokenized_articles_lock:
+                            num_tokenized_articles.value += 1
+                except json.JSONDecodeError as exc:
+                    if ignore_input_format_error:
+                        with open(error_log_path, "a") as f:
+                            f.write(line)
+                        if num_tokenized_articles_lock is not None and num_skipped_articles is not None:
+                            with num_tokenized_articles_lock:
```

Comment:
I had the same question started out and tried to search it but didn't find a clear answer. But I think it is fine since the lock object and the variable objects are not referencing each other. They are all created from the `manager`.